tracer/test/Datadog.Trace.SourceGenerators.Tests/AspectsDefinitionsGeneratorTests.cs

```diff
@@ -0,0 +1,668 @@
+// <copyright file="AspectsDefinitionsGeneratorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.SourceGenerators.AspectsDefinitions;
+using Datadog.Trace.SourceGenerators.InstrumentationDefinitions;
+using Datadog.Trace.SourceGenerators.InstrumentationDefinitions.Diagnostics;
+using Xunit;
+
+namespace Datadog.Trace.SourceGenerators.Tests
+{
+    public class AspectsDefinitionsGeneratorTests
+    {
+        [Fact]
+        public void DoesNotGenerateDefinitionsIfThereAreNone()
+        {
+            const string input = """
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.Iast;
+using Datadog.Trace.Iast.Dataflow;
+
+namespace MyTests;
+public class TestList 
+{ 
+}
+""";
+
+            const string expected = """
+// <auto-generated/>
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    internal static partial class AspectDefinitions
+    {
+        public static string[] Aspects = new string[] {
+        };
+    }
+}
+
+""";
+
+            var (diagnostics, output) = TestHelpers.GetGeneratedOutput<AspectsDefinitionsGenerator>(
+                SourceHelper.AspectAttributes,
+                input);
+
+            Assert.Equal(expected, output);
+            Assert.Empty(diagnostics);
+        }
+
+        [Fact]
+        public void CanGenerateAspectsDefinitionForSimpleClass()
+        {
+            const string input = """
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.Iast;
+using Datadog.Trace.Iast.Dataflow;
+
+namespace MyTests;
+
+[AspectClass("mscorlib,netstandard,System.Private.CoreLib", AspectFilter.StringOptimization)]
+public class TestAspectClass
+{ 
+    [AspectMethodReplace("System.String::Concat(System.String,System.String)", AspectFilter.StringLiterals_Any)]
+    public static string Concat(string target, string param1)
+    {
+        return string.Concat(target, param1);
+    }
+    [AspectMethodReplace("System.String::Concat(System.Object,System.Object,System.Object)", AspectFilter.StringLiterals_Any)]
+    public static string Concat(object target, object param1, object param2)
+    {
+        return string.Concat(target, param1, param2);
+    }
+}
+""";
+
+            const string expected = """"
+// <auto-generated/>
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    internal static partial class AspectDefinitions
+    {
+        public static string[] Aspects = new string[] {
+"""[AspectClass("mscorlib,netstandard,System.Private.CoreLib",[StringOptimization],Propagation,[])] MyTests.TestAspectClass""",
```

Comment:
As with my previous note, this is what it was generating, but it's not valid C#. It needs to be something like

```csharp
"""
[AspectClass("mscorlib,netstandard,System.Private.CoreLib",[StringOptimization],Propagation,[])] MyTests.TestAspectClass
"""
```

With the `"""` on different lines, and the indent matching.

You can test it though - copy this string to a new file and see if it errors. If it's fine, then I'm wrong!

Reply:
Fixed